### PR TITLE
Integrates SimplenoteFoundation

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're integrating SimplenoteFoundation. This new Swift Package is to be known as the home of the following components:

- ResultsController
- EntityObserver
- And a ton of unit tests

### CircleCI:
I've ran into problems with CircleCI. **xcodebuild** was unable to clone the Swift Packages dependency (SSH issue).

- [This workaround has proven to be effective]( https://support.circleci.com/hc/en-us/articles/360044709573-Swift-Package-Manager-fails-to-clone-from-private-Git-repositories). 
- I also had to bump up Xcode to 11.6 (11.2 caused a segmentation fault!!)

cc @mokagio / @jkmassel gentlemen WDYT, do you consider this is properly addressed? (`.circleci` [setup edited here](https://github.com/Automattic/simplenote-ios/pull/843/files#diff-1d37e48f9ceff6d8030570cd36286a61)!)
Thank you!

### Test
- [ ] Verify the app builds correctly
- [ ] Run a smoke test and verify the Notes List works as expected!

### Release
These changes do not require release notes.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,16 +34,21 @@ jobs:
       - run:
           name: Copy demo SPCredentials
           command: mkdir -p Simplenote/Credentials && cp Simplenote/SPCredentials-demo.swift Simplenote/Credentials/SPCredentials.swift
+      # Start: Swift Package Manager Workaround
+      # Ref. https://support.circleci.com/hc/en-us/articles/360044709573-Swift-Package-Manager-fails-to-clone-from-private-Git-repositories
```
Comment (jkmassel) [inline .circleci/config.yml:38]: cool.gif

File: Simplenote.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "object": {
+    "pins": [
+      {
+        "package": "SimplenoteFoundation",
+        "repositoryURL": "git@github.com:Automattic/SimplenoteFoundation-Swift.git",
+        "state": {
+          "branch": "issue/visibility-modifiers",
```
Comment (jkmassel) [inline Simplenote.xcworkspace/xcshareddata/swiftpm/Package.resolved:8]: Do we want this pointing at a branch in `develop`? Or should it perhaps be versioned like a Pod?

Comment (jleandroperez) [reply Simplenote.xcworkspace/xcshareddata/swiftpm/Package.resolved:8]: Oh! It'll be pointing to actual released versions. It's mapping towards a specific branch for testing purposes, I'll switch before merging.

Thank you for the heads up sir!

**Comment by jleandroperez** [comment]
Thank you @jkmassel !!! Wooooo!!

**Comment by jleandroperez** [comment]
For the record!

- SwiftPackagesManager usage caused Xcode to bump up an objectVersion in our **xcodeproj**
- Because of the above, Cocoapods started moving several strings from **xcodeproj** over to a plain textfile
- Our caching prevents regenerating such a file

I've bumped up the Tracks dependency, in order to force the Cocoapods cache to be invalidated.

Thanks everyone!!

**Pull request merged**